### ISSUE TITLE
Storage doesn't decrease after deleting media

### ISSUE THREAD

**Issue by jeroenops** [mollyim/mollyim-android#21]
**Thank you**

First of all, a sincere thank you for the continued development of Molly. I deeply appreciate the effort to keep us safe. Especially with the recently added FOSS version, Molly has now multiple major advantages (IMHO).

**Issue**

I've had more than a years worth of conversations and media stored in Molly. To start the new year properly, I decided to clean it up a bit and set the maximum number of messages in conversations to 250 and to retain conversations for six months. Both gave a pop-up that it would start pruning old messages and media right away. Confirmed both. However, it didn't shrunk the reported storage size - still at 3,7GB. I've went through the storage manager; the largest file is 19MB, with the majority at a few hundred KB. It should equal no more than 1GB at max, likely far less.

Until now, two days went by, issued a reboot in the meantime, but the storage doesn't decrease. In fact, it went up from 3,7GB to 3,8GB after receiving a couple of videos from a relative. Android itself reports Molly to have user data totalling 3,8GB.

Just made a backup in Molly, the backup file is 708MB.

Could this be a bug? Something I can do to help you investigate this?

**Comment by valldrac** [comment]
Thanks for reporting it. It is definitely a bug. There are some known issues with Signal storage. See this ticket for example https://github.com/signalapp/Signal-Android/issues/10179

It could also be a bug in Molly. Less probable but it cannot be ruled out. Do you have the passphrase encryption and the autolock enabled?

If you're familiar with adb and USB debugging options, you can get root access and see what's going on in the internal storage of the app, at the path `/data/data/im.molly.app`. Then I can give you more ideas, but first you need to get access.

**Comment by jeroenops** [comment]
Sincere apologies for the late reply.

I run Molly on a GrapheneOS device, logically, without root enabled. And since rooting it means the data gets wiped, it seems counterproductive toward producing a log. However, I seem to have fixed the issue.

The outage that Signal suffered from yesterday was made clear through a 'notification banner' (mentioning that they were experiencing technical difficulties). However, after it was fixed earlier today, the banner did not disappear on my Molly app, even though friends reported the banner was gone for them.

I decided to make a backup and reinstall Molly, recovering the backup afterwards. That seems to have fixed both the persistent banner and the reported storage usage. I have no clue as to whether this problem is Molly-specific or something upstream (in Signal). My assumption is the latter :)
